### PR TITLE
ZTE-SH-CN-run-application-run-single-instance-stateful-application-pr-2017-…

### PR DESCRIPTION
**Pull request by jianglingxia**
…09-09

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.8 Features: set Milestone to `1.8` and Base Branch to `release-1.8`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5363)

### CODE REVIEW

**Comment by jianglingxia** [comment]
cc @zhangxiaoyu-zidif

**Comment by markthink** [comment]
@jianglingxia Please close PR, update to k8smeetup library, unified submission..Do not submit a single file..

update table:
https://docs.google.com/spreadsheets/d/1FDFCv9RK5nSMgLXhPrJ5k7r5QvHnNEFnXbvoFiM8v20/edit#gid=1015862274

**Comment by zhangxiaoyu-zidif** [comment]
@markthink 
I think we need not close this pr. 
And we do not have authority  to modify the docs you mentioned.
we can review the pr here until it could be merged. 
One markdown file，one PR， is there any problem?

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  Pod. For clustered stateful apps, see the
+  [StatefulSet documentation](/docs/concepts/workloads/controllers/petset/).
+* Use `strategy:` `type: Recreate` in the Deployment configuration
+  YAML file. This instructs Kubernetes to _not_ use rolling
+  updates. Rolling updates will not work, as you cannot have more than
+  one Pod running at a time. The `Recreate` strategy will stop the
+  first pod before creating a new one with the updated configuration.
+-->
+
+Deployment中镜像或其他部分照样可以通过 `kubectl apply` 命令更新. 以下是特定于
```
Comment (tengqm) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: “照样”过于口语化，且引起误解。是照什么样，哪个样？

as usual，是”同往常一样“

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+有状态应用的一些预防措施:
```
Comment (tengqm) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: precaution翻译成”注意事项“更合适

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+* 不要弹性伸缩. 更新仅适用于单实例应用. PersistentVolume的底层仅只能挂载一个pod. 对于集群级有状态应用, 请参考
```
Comment (tengqm) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: setup为什么翻译成”更新"?

the underlying PersistentVolume是说下层的PersistentVolume，不是说PersistentVolume的底层

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  [StatefulSet documentation](/docs/concepts/workloads/controllers/petset/).
```
Comment (tengqm) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: Stateful doucmentation -> StatefulSet 文档

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+```
+gcloud compute disks delete mysql-disk
+```
+
+{% endcapture %}
+
+
+{% capture whatsnext %}
+
+* 了解更多 [Deployment objects](/docs/concepts/workloads/controllers/deployment/).
```
Comment (tengqm) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: 翻译链接。下同。

Comment (zhangxiaoyu-zidif) [reply cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: 这里应该使用/cn/...的链接。

**Comment by jianglingxia** [comment]
thanks very much @tengqm ,I have updated it!

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 6]
```diff
@@ -62,14 +61,13 @@ for details on writing a PersistentVolume configuration file for other
 environments.
 -->
 
-注意`pdName: mysql-disk` 这行与计算引擎环境中的磁盘名称相匹配. 有关为其他环境
-编写PersistentVolume配置文件的详细信息，请参见
-[Persistent Volumes](/docs/concepts/storage/persistent-volumes/).
+注意`pdName: mysql-disk` 这行与Compute Engine环境中的磁盘名称相匹配. 有关为其
+他环境编写PersistentVolume配置文件的详细信息，请参见持久卷[Persistent Volumes](/docs/concepts/storage/persistent-volumes/).
 <!--
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: 此处注释删除。再试试能否检查通过。

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 7]
```diff
@@ -62,14 +61,13 @@ for details on writing a PersistentVolume configuration file for other
 environments.
 -->
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: 此处注释删除

**Comment by zhangxiaoyu-zidif** [comment]
@jianglingxia 
gernelly is read well, so please delete English comment with <!-->and then let me see if the deploy/netlify works well.

**Comment by jianglingxia** [comment]
@zhangxiaoyu-zidif I have updated it,and all checks have passed!thanks

**Comment by tengqm** [comment]
Please check the preview page https://deploy-preview-5363--kubernetes-io-master-staging.netlify.com/cn/docs/tasks/run-application/run-single-instance-stateful-application/

You will see how it looks like and you will realize there are quite some places you want to polish.

**Comment by jianglingxia** [comment]
is it Browser problem? I use the firefox is ok ,and I check my md file compared with the docs/task/run-application file,the result is both same! @tengqm @zhangxiaoyu-zidif

**Comment by jianglingxia** [comment]
image

**Comment by jianglingxia** [comment]
can label approve this pr？ @tengqm  @zhangxiaoyu-zidif 
thanks

**Comment by zhangxiaoyu-zidif** [comment]
@steveperry-53 
LGTM

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+{% endcapture %}
+
+
+{% capture prerequisites %}
+
+* {% include task-tutorial-prereqs.md %}
+
+* 为了数据持久性我们将在环境上通过磁盘创建一个持久卷. 环境支持的类型见这里[here](/docs/user-guide/persistent-volumes/#types-of-persistent-volumes). 本篇文档将介绍 `GCEPersistentDisk` . `GCEPersistentDisk`卷只能工作在Google Compute Engine
+平台上.
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: "平台上"应该与上一行一起是一行，不用换行。

File: cn/docs/tasks/run-application/run-single-instance-stateful-application.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+```
+
+## 更新
+
+
+Deployment中镜像或其他部分同往常一样可以通过 `kubectl apply` 命令更新. 以下是
+特定于有状态应用的一些注意事项:
+
+* 不要弹性伸缩. 弹性伸缩仅适用于单实例应用. 下层的PersistentVolume仅只能挂载一个pod. 对于集群级有状态应用, 请参考StatefulSet文档
+  [StatefulSet documentation](/docs/concepts/workloads/controllers/petset/).
```
Comment (zhangxiaoyu-zidif) [inline cn/docs/tasks/run-application/run-single-instance-stateful-application.md]: 不要弹性伸缩. 弹性伸缩仅适用于单实例应用. 下层的PersistentVolume仅只能挂载一个pod. 对于集群级有状态应用, 请参考[StatefulSet文档](/docs/concepts/workloads/controllers/petset/).
此处再改为StatefulSet文档。
顺便把英文的链接也改了。

**Comment by zhangxiaoyu-zidif** [comment]
please fix left minor flaws...
Then LGTM.

**Comment by zhangxiaoyu-zidif** [comment]
@steveperry-53 
LGTM

**Pull request merged**